x-pack/plugins/siem/server/utils/build_validation/exact_check.ts

```diff
@@ -47,7 +44,7 @@ export const exactCheck = <T>(
   return pipe(decoded, fold(onLeft, onRight));
 };
 
-export const findDifferencesRecursive = <T>(original: object, decodedValue: T): string[] => {
+export const findDifferencesRecursive = <T>(original: T, decodedValue: T): string[] => {
```

Comment:
This type reads a little weird without a concept of subtypes, but this is more accurate than before 👍